src/interpreter/Opcode.cpp

```diff
@@ -17,16 +17,19 @@
 #include "Walrus.h"
 
 #include "interpreter/Opcode.h"
+#include "interpreter/ByteCode.h"
 
 namespace Walrus {
 
+// Note: if size is not initialized in opcode.def then this will cause a compile error
 ByteCodeInfo g_byteCodeInfo[OpcodeKind::InvalidOpcode] = {
 #define WABT_OPCODE(rtype, type1, type2, type3, memSize, prefix, code, name, \
-                    text, decomp)                                            \
+                    text, decomp, size)                                      \
```

Comment:
This patch insert additional size info to the global table (`g_byteCodeInfo`)
and this approach would require more memory for the global table.

I'm not sure that it is a better way to get a size of bytecode.
I also wonder the main purpose of this patch. Is there any special reason for this?

Reply:
The aim is allow iterating through the byte code in release mode. Currently jit only works in debug mode because of that. And future byte code optimizations could also do that.

We could make a separate uint8_t array for the sizes, that would remove the 7 byte alignment loss in `g_byteCodeInfo`.

We are also open to other ideas.

Reply:
https://github.com/Samsung/walrus/blob/175045da7830043dcf01d6ddf7b7b7f70a0980e5/src/interpreter/Interpreter.cpp#L594-L607

@zherczeg 
Our interpreter needs to iterate each bytecode too.
For this case, as you can find from the above code (`BrTable`) in interpreter, interpreter calculates the size of bytecode when its required.
I think that JIT code could calculate the bytecode size similarly, but is there any issue for this?
Only 5 bytecodes require calculation of their size (End, Throw, Call, CallIndirect, BrTable)

Reply:
its just a proposal.
If JIT should iterate bytecode stream any time it needs,
what about storing the full size of bytecode in the variable-sized bytecode?

For example, when parser generates a `Throw` bytecode, it calculates the total size of the bytecode and stores the size within the created `Throw` bytecode together.
This additional size info is required only for 5 variable-sized bytecodes  (End, Throw, Call, CallIndirect, BrTable)

Reply:
Yes, please change it if you all are fine with this :)
IMHO its better to apply simple ways first and optimize it later

Reply:
As soon as pr #86 is in I will be updating this patch, seeing that it is directly linked to this problem.